src/libstd/sys/unix/fs.rs

```diff
@@ -132,6 +139,32 @@ impl FromInner<raw::mode_t> for FilePermissions {
 impl Iterator for ReadDir {
     type Item = io::Result<DirEntry>;
 
+    #[cfg(target_os = "sunos")]
```

Comment:
Interesting! In that case if we need a whole separate implementation *anyway* then I'm fine sticking with `readdir` as a separate implementation. Can you add some documentation indicating that although this function is avoided on Linux it's ok to call on sunos?